### ISSUE TITLE
Zathura synctex support

### ISSUE THREAD

**Issue by Vintodrimmer** [voidlinux/void-packages#7022]
Good day,

Consider enabling `synctex` support for the `zathura`, unless there are some reasons not to.

In particular, it is needed for the correct `vimtex` integration. https://github.com/lervag/vimtex/issues/901#issuecomment-314577997

**Comment by chneukirchen** [comment]
This will require packaging libsynctex first, which we use as a vendored binary from TeXLive currently.

**Comment by Vintodrimmer** [comment]
Got it.

Is there a way for me to compile it with the support, if I have `synctex` installed through `tlmgr`?
Maybe something to change in the template?

**Comment by stuart-little** [comment]
The way I got around this eventually is not ideal, but it works for now: 

I got the rpms for [libsynctex](https://www.rpmfind.net/linux/rpm2html/search.php?query=libsynctex.so.1()(64bit)) and [its devel version](http://rpmfind.net/linux/rpm2html/search.php?query=pkgconfig(synctex)) respectively, extracted them with rpmextract and distributed the resulting files in the right spots in /usr, then compiled zathura from source (make + make install). 

I needed pkg-config to find the libsynctex files, because that's how zathura knows whether libsynctex is on your system.  

I first tried to go [the xbps-src route](https://wiki.voidlinux.eu/Xbps-src), but when installing zathura that way after setting up libsynctex, for some reason zathura would *still* build without support. I am not sure why this is; when I do 'make' manually in the zathura source folder it sees libsynctex. 

I'm sure I'm missing something obvious; I'm new to void.

**Comment by chneukirchen** [comment]
I think adding libsynctex should be straight forward, did you add it to makedepends?

**Comment by stuart-little** [comment]
I did not. That does sound very simple if it's all it'll take, but my excuse is it was literally my first try with xbps-src. I've been using void for only days (forget how many; certainly under a week). 

Manual installs are annoying because they leave the software outside of xbps's database, so when I want to install zathura plugins I can't do that through xbps either (because it thinks zathura isn't on my system). So I would like to do this through xbps, and have a few options I'd like to test: 

- what you suggest, makedepends or whatever other means of altering the xbps-src pkg make process I can dig up from the documentation;

- make+make install to a separate directory and then run xbps-xreate; I learned about this from lurking on the forum, and seems like a good hybrid solution. 

I'll do some experimentation and report back if / when it works out.

**Comment by chneukirchen** [comment]
Oh, I see the problem now, synctex is only in texlive and not so easy to build manually.

**Comment by stuart-little** [comment]
Precisely, yes. libsynctex isn't a standalone package in this case, it's a bunch of files I manually extracted out of two rpm's and scattered in the right places on my system (as described briefly in my first message).

The second option from my previous reply worked though, so I'll settle on it as a stable solution for now (I have to actually do some work with these packages I keep fiddling with :) ):

(1) I ran 'make' on zathura myself so as to have it find the various libsynctex files on my system;

(2) ran sudo make DESTDIR={some isolated directory of my choosing} install;

(3) ran xbps-create (options, short desc., architecture, etc.) on the directory from the previous step;

(4) ran sudo xdowngrade on the resulting .xbps file. 

Worked like a charm: I now have zathura with synctex support that xbps knows about, and was then able to install zathura plugins directly through xbps without having to make-install them myself. This allows me to keep the manually-compiled material on my system at a minimum and have the package manager handle as much of it as it can.  

Tested it just now and back-forth sync works just as they should.

**Comment by svenper** [comment]
FWIW, I have a [WIP template](/svenper/home-bin/tree/master/patches/templates/texlive2017-exec) for building binaries and dynamic libraries from texlive sources, with a subpackage for libsynctex among others.

**Comment by xaltsc** [comment]
@svenper Sorry to dig this issue up from last year, but I've found no other way of contacting you. Have you managed to build libsynctex-devel since ? I cloned your home-bin repo on gitlab and I cant manage to ./xbps-src pkg texlive2018-exec-libsynctex-devel, the build fails at some point where a type GBool is unkown to the compiler.
